python/Ganga/GPIDev/Lib/Job/Job.py

```diff
@@ -1472,6 +1471,10 @@ def submit(self, keep_going=None, keep_on_fail=None, prepare=False):
             # split into subjobs
             rjobs = self._doSplitting()
 
+            # Some old jobs may still contain None assigned to the self.subjobs so be careful when checking length
+            if self.subjobs is not None and len(self.subjobs) == 0 and self.splitter is not None:
```

Comment:
Even before the default was changed it was set to `[]` I think which is safe to do a `len()` on? Regardless, as this code is run on `submit()`, the chance of someone having a job in `new` from that long ago which they have not submitted yet is very small.

It should be as simple as:
```python
if self.splitter is not None and not self.subjobs:
    raise ...
```